File: Array | Dictionary | Set/replace_elements_with_greatest_element_on_right_side/js/replace_elements_with_greatest_element_on_right_side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    let result = [];
    let length = arr.length;
    let maxnum = -1;

    for ( let i = length - 2; i >=0 ; i-- ) {
        maxnum = Math.max(maxnum,arr[i+1]);
        result[i] = maxnum
    }

    result[length-1] = -1;
    return result;
};

File: Array | Dictionary | Set/replace_elements_with_greatest_element_on_right_side/py/replace_elements_with_greatest_element_on_right_side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        result = [i for i in range(len(arr))]
        maxnum = -1
        
        for i in range(len(arr)-2,-1,-1):
            maxnum = max(maxnum,arr[i+1])
            result[i] = maxnum
        
        result[len(arr)-1] = -1

        return result